src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/TargetFrameworkProjectConfigurationDimensionProvider.cs

```diff
@@ -31,16 +31,15 @@ public TargetFrameworkProjectConfigurationDimensionProvider(IProjectXmlAccessor
         {
             Requires.NotNull(project, nameof(project));
 
-            string targetFrameworkProperty = await _projectXmlAccessor.GetEvaluatedPropertyValue(project, ConfigurationGeneral.TargetFrameworkProperty).ConfigureAwait(true);
-            if (targetFrameworkProperty != null)
+            string targetFrameworksProperty = await _projectXmlAccessor.GetEvaluatedPropertyValue(project, ConfigurationGeneral.TargetFrameworksProperty).ConfigureAwait(true);
+            if (targetFrameworksProperty != null)
             {
-                // If the project already defines a specific "TargetFramework" to target, then this is not a cross-targeting project and we don't need a target framework dimension.
-                return ImmutableArray<string>.Empty;
+                return BuildUtilities.GetPropertyValues(targetFrameworksProperty);
             }
             else
             {
-                string targetFrameworksProperty = await _projectXmlAccessor.GetEvaluatedPropertyValue(project, ConfigurationGeneral.TargetFrameworksProperty).ConfigureAwait(true);
-                return BuildUtilities.GetPropertyValues(targetFrameworksProperty);
+                // If the project doesn't have a "TargetFrameworks" property, then this is not a cross-targeting project and we don't need a target framework dimension.
+                return ImmutableArray<string>.Empty;
```

Comment:
>return ImmutableArray<string>.Empty; [](start = 16, length = 36)

You may want to double check that this isn't the default behavior of the base class. If it is you can completely remove the overload.